dockerfiles/e2e-saas/docker-entrypoint.sh

```diff
@@ -0,0 +1,148 @@
+# ------------------------------------------------------------------------------------------
+
+echo "Trying to find token for $(echo $USERNAME | cut -c1-3) $(echo $USERNAME | cut -c4-$length)"  
+    
+#verify environment - if production or prod-preview
+#variable preview is used differ between prod and prod-preview urls
+if [[ "$USERNAME" == *"preview"* ]]; then
+	response=$(curl -s -g -X GET --header 'Accept: application/json' "https://api.prod-preview.openshift.io/api/users?filter[username]=$USERNAME")
+	echo $response
+	data=$(echo "$response" | jq .data)
+	if [ "$data" == "[]" ]; then
+		echo -e "${RED}User $USERNAME is not provisoned on prod-preview cluster. Please check settings. ${NC}"
+	    exit 1
+    fi        
+    preview="prod-preview."
+else
+	response=$(curl -s -g --header 'Accept: application/json' -X GET "https://api.openshift.io/api/users?filter[username]=$USERNAME")
+	data=$(echo "$response" | jq .data)
+	if [ "$data" == "[]" ]; then
+		echo -e "${RED}User $USERNAME is not provisioned on production cluster. Please check settings. ${NC}" 
+        exit 1
+    fi        
+	preview=""
+fi
+		
+#get html of developers login page
+curl -sX GET -L -c cookie-file -b cookie-file "https://auth.${preview}openshift.io/api/login?redirect=https://che.openshift.io" > loginfile.html
+
+#get url for login from form
+url=$(grep "form id" loginfile.html | grep -o 'http.*.tab_id=.[^\"]*')
+dataUrl="username=$USERNAME&password=$PASSWORD&login=Log+in"
+url=${url//\&amp;/\&}
+
+#send login and follow redirects  
+set +e
+url=$(curl -w '%{redirect_url}' -s -X POST -c cookie-file -b cookie-file -d "$dataUrl" "$url")
+found=$(echo "$url" | grep "token_json")
+
+while true 
+do
+	url=$(curl -c cookie-file -b cookie-file -s -o /dev/null -w '%{redirect_url}' "$url")
+	if [[ ${#url} == 0 ]]; then
+		#all redirects were done but token was not found
+		break
+	fi
+	found=$(echo "$url" | grep "token_json")
+	if [[ ${#found} -gt 0 ]]; then
+		#some redirects were done and token was found as a part of url
+		break
+	fi
+done
+set -e
+
+#substract active token
+token=$(echo "$url" | grep -o "ey.[^%]*" | head -1)
+if [[ ${#token} -gt 0 ]]; then
+    #save each token into file tokens.txt in format: token;username;["","prod-preview"]
+	export E2E_SAAS_TESTS_USER_TOKEN=${token}
+	echo "Token set successfully."
+else
+	echo -e "${RED}Failed to obtain token for $USERNAME! Probably user password is incorrect. Continue with other users. ${NC}"
+	exit 1
+fi
+
+# ------------------------------------------------------------------------------------------
+# -------------------- SET ALL NEEDED ENVIRONMENT VARIABLES --------------------------------
+# ------------------------------------------------------------------------------------------
+
+export TS_SELENIUM_USERNAME=$USERNAME
+export TS_SELENIUM_PASSWORD=$PASSWORD
+export TS_SELENIUM_BASE_URL=$URL
+#E2E_SAAS_TESTS_USER_TOKEN is already set
```

Comment:
do we need this comment? it does not clarify things for me personally

Reply:
Comment removed.